images/router/haproxy/conf/haproxy-config.template

```diff
@@ -377,7 +380,7 @@ backend be_secure:{{$cfgIdx}}
   {{- end }}
 
   {{- if not (isTrue (index $cfg.Annotations "haproxy.router.openshift.io/disable_cookies")) }}
-  cookie {{$cfg.RoutingKeyName}} insert indirect nocache httponly
+  cookie {{firstMatch $cookieName (index $cfg.Annotations "router.openshift.io/route.cookie.name") $cfg.RoutingKeyName}} insert indirect nocache httponly
```

Comment:
The annotation should probably be:
  haproxy.router.openshift.io/cookie_name

Or cookie-name... we have both in the conf file :-(  Seems redundant to say route there though since it is on the route.

Also, you lost the global env... do we want to have that as an option?

Reply:
Went with cookie_name

Reply:
Added environment variable ROUTER_COOKIE_NAME